src/communication/communicator.hpp

```diff
@@ -105,16 +106,15 @@ class communicator {
     /// group as a result of the global spike exchange.    
     std::vector<event_queue> make_event_queues(const gathered_vector<spike_type>& global_spikes)
     {
+        // turn pair<it1, it2> into a class with begin()/end()
+        using nest::mc::util::make_range;
+        using nest::mc::util::lessthan;
         // Comparator operator between a spike and a spike source for equal_range
-        struct comp {
+        struct extractor {
```

Comment:
Here's the other approach: I added a lessthan utility class that makes the comparisons, and then you just need to define extractor to return what you actually want to compare: here, the id_type.